frontend/dockerfile/dockerfile2llb/convert_test.go

```diff
@@ -27,6 +27,57 @@ func toEnvMap(args []instructions.KeyValuePairOptional, env []string) map[string
 	return m
 }
 
+func TestDockerfileLinting(t *testing.T) {
```

Comment:
We can either add both unit and integration tests for the rules or only integration tests. I see in the other PR you started to add integration tests as well.

Ideally we make it so that a test case has a Dockerfile, and expected warnings. Then integration test variant would build it and check for expected warnings in progress (we should have existing test for that can be modified) and another variant does the same via `--print=lint` and checks the expected warnings in output.